lib/ansible/plugins/lookup/template.py

```diff
@@ -92,7 +92,15 @@ def run(self, terms, variables, **kwargs):
                     searchpath = newsearchpath
                 searchpath.insert(0, os.path.dirname(lookupfile))
 
-                self._templar.environment.loader.searchpath = searchpath
+                self._templar.environment.loader.searchpath = j2_searchpath = []
+                for p in searchpath:
+                    j2_searchpath.append(
+                        to_native(
```

Comment:
We should probably put a comment about why we do any double encoding/decodings when we do it.  This one is particularly tricky because it deals with jinja2's expectations and native strings instead of either bytes or text.  So it probably deserves some explicit spelling out of the different cases we expect to encounter.